algosdk/abi/method.py

```diff
@@ -0,0 +1,219 @@
+        method_dict = json.loads(resp)
+        return Method.undictify(method_dict)
+
+    @staticmethod
+    def from_signature(s):
+        # Split string into tokens around outer parentheses.
+        # The first token should always be the name of the method,
+        # the second token should be the arguments as a tuple,
+        # and the last token should be the return type (or void).
+        tokens = Method._parse_string(s)
+        argument_list = [
+            Argument(t) for t in abi.TupleType._parse_tuple(tokens[1])
+        ]
+        return_type = Returns(tokens[-1])
+        return Method(name=tokens[0], args=argument_list, returns=return_type)
+
+    def dictify(self):
+        d = {}
+        d["name"] = self.name
+        d["args"] = [arg.dictify() for arg in self.args]
+        if self.returns:
+            d["returns"] = self.returns.dictify()
+        if self.desc:
+            d["desc"] = self.desc
+        return d
+
+    @staticmethod
+    def undictify(d):
+        name = d["name"]
+        arg_list = [Argument.undictify(arg) for arg in d["args"]]
+        return_obj = (
+            Returns.undictify(d["returns"]) if "returns" in d else None
+        )
+        desc = d["desc"] if "desc" in d else None
+        return Method(name=name, args=arg_list, returns=return_obj, desc=desc)
+
+
+class Argument:
+    """
+    Represents an argument for a ABI method
+
+    Args:
+        arg_type (string): ABI type or transaction string of the method argument
+        name (string, optional): name of this method argument
+        desc (string, optional): description of this method argument
+    """
+
+    def __init__(self, arg_type, name=None, desc=None) -> None:
+        if arg_type in TRANSACTION_ARGS:
+            self.type = arg_type
+        else:
+            # If the type cannot be parsed into an ABI type, it will error
+            self.type = abi.util.type_from_string(arg_type)
+        self.name = name
+        self.desc = desc
+
+    def __eq__(self, o: object) -> bool:
+        if not isinstance(o, Argument):
+            return False
+        return (
+            self.name == o.name and self.type == o.type and self.desc == o.desc
+        )
+
+    def __str__(self):
+        return str(self.type)
+
+    def dictify(self):
+        d = {}
+        d["type"] = str(self.type)
+        if self.name:
+            d["name"] = self.name
+        if self.desc:
+            d["desc"] = self.desc
+        return d
+
+    @staticmethod
+    def undictify(d):
+        return Argument(
+            arg_type=d["type"],
+            name=d["name"] if "name" in d else None,
+            desc=d["desc"] if "desc" in d else None,
+        )
+
+
+class Returns:
+    """
+    Represents a return type for a ABI method
+
+    Args:
+        arg_type (string): ABI type of this return argument
+        desc (string, optional): description of this return argument
+    """
+
+    def __init__(self, arg_type, desc=None) -> None:
```

Comment:
I believe "void" deserves it's own constant -at least- so it'll be easier for the linter to pick up any errors and avoid the hard-coded string "void" in other locations. For example, you could have something like
```python
class Returns:
    """ ... 
    """
    VOID = "void"
```
and below:
```python
        if arg_type == "void":
            self.type = self.VOID
```

Reply:
That look nice, I'll do that thanks